networkx/algorithms/shortest_paths/tests/test_generic.py

```diff
@@ -324,13 +324,17 @@ def test_specified_methods(self):
         )
         assert ans == pytest.approx(4, abs=1e-7)
 
-    def test_disconnected(self):
+    def test_directed_not_strongly_connected(self):
+        g = nx.DiGraph()
+        g.add_nodes_from(range(2))
+        g.add_edge(0, 1)
+        pytest.raises(nx.NetworkXError, nx.average_shortest_path_length, g)
```

Comment:
Mostly cosmetic suggestion, not a blocker:

```suggestion
    def test_directed_not_strongly_connected(self):
        G = nx.DiGraph([(0, 1)])
        with pytest.raises(nx.NetworkXError, match="Graph is not strongly connected"):
            nx.average_shortest_path_length(G)
```